packages/common/src/services/AccessService.ts

```diff
@@ -0,0 +1,51 @@
+import { inject, injectable } from 'inversify';
+
+import type { Passport } from '../../types/passport';
+
+import StorageService from './StorageService';
+
+const PASSPORT_KEY = 'passport';
+
+@injectable()
+export default class AccessService {
+  private readonly storageService: StorageService;
+
+  constructor(@inject(StorageService) storageService: StorageService) {
+    this.storageService = storageService;
+  }
+
+  generatePassport = async (host: string, siteId: string, jwt?: string): Promise<Passport | null> => {
+    if (!siteId) {
+      throw new Error('Site ID is required');
+    }
+
+    const pathname = `/v2/sites/${siteId}/access/generate`;
+    const url = `${host}${pathname}`;
+
+    const response = await fetch(url, {
+      method: 'PUT',
+      headers: {
+        Authorization: jwt ? `Bearer ${jwt}` : '',
+      },
+    });
+
+    if (!response.ok) {
+      return null;
+    }
+
+    return (await response.json()) as Passport;
+  };
+
+  setPassport = (passport: Passport) => {
```

Comment:
Was there a reason you put the storage (set/get/remove) in the access service instead of in the access controller? It seems like it might be cleaner to have a controller that calls 2 services instead of a controller that calls a service that calls a service. Thoughts?